OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -293,9 +310,12 @@ public virtual IEnumerable<ActorInfo> BuildableItems()
 				return Enumerable.Empty<ActorInfo>();
 			if (!Enabled)
 				return Enumerable.Empty<ActorInfo>();
-			if (developerMode.AllTech)
+			if (!Info.PayUpFront && developerMode.AllTech)
 				return Producible.Keys;
-
+			if (Info.PayUpFront && developerMode.AllTech)
+				return Producible.Keys.Where(a => GetProductionCost(a) < playerResources.GetCashAndResources() || IsInQueue(a));
+			if (Info.PayUpFront)
+				return buildableProducibles.Where(a => GetProductionCost(a) < playerResources.GetCashAndResources() || IsInQueue(a));
```

Comment:
```suggestion
				return buildableProducibles.Where(a => GetProductionCost(a) <= playerResources.GetCashAndResources() || IsInQueue(a));
```

otherwise you can't queue assets that cost exactly the amount of cash you have